src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -602,8 +639,57 @@ def post_process(self, r):
                 linker_client.linker.begin_create_or_update(resource_uri=r["id"],
                                                             parameters=item["parameters"],
                                                             linker_name=item["linker_name"]).result()
+
+        if self.get_argument_repo():
+            r = self.set_up_create_containerapp_if_source_or_repo()
+
         return r
 
+    def set_up_create_containerapp_if_source_or_repo(self):
+        from ._up_utils import (ContainerApp, ResourceGroup, ContainerAppEnvironment, _reformat_image, get_token, _has_dockerfile, _get_dockerfile_content, _get_ingress_and_target_port, _get_registry_details, _create_github_action, get_token)
+        ingress = self.get_argument_ingress()
+        target_port = self.get_argument_target_port()
+        dockerfile = "Dockerfile"
+        token = get_token(self.cmd, self.get_argument_repo(), self.get_argument_token())
+        # Parse resource group name and managed env name
+        env_id = self.containerapp_def["properties"]['environmentId']
+        parsed_managed_env = parse_resource_id(env_id)
+        env_name = parsed_managed_env['name']
+        env_rg = parsed_managed_env['resource_group']
+
+        # Parse location
+        env_info = self.get_environment_client().show(cmd=self.cmd, resource_group_name=env_rg, name=env_name)
+        location = env_info['location']
+
+        # Set image to None if it was previously set to the default image (case where image was not provided by the user) else reformat it
+        image = None if self.get_argument_image().__eq__(HELLO_WORLD_IMAGE) else _reformat_image(self.get_argument_source(), self.get_argument_repo(), self.get_argument_image())
+
+        if not self.get_argument_source() or _has_dockerfile(self.get_argument_source(), dockerfile):
```

Comment:
nit: can we assign the value of `_has_dockerfile(self.get_argument_source(), dockerfile)` to a variable since we use the same expression down below on line 680 when calling `app.run_acr_build` for the `build_from_source` argument?

Reply:
Updated.